### PR TITLE
Refactoring for nautilus branch

### PR DESCRIPTION
**Pull request by yuezhu**
Some refactoring changes for nautilus branch, including:
* Upgrade go version to 1.15.3
* Upgrade `prometheus_client` to 1.8.0
* Upgrade `go-ceph` to 0.6.0
* Migrate to `go module`
* Use `logrus` for structured logging and level control
* Use environment variable to pass in parameters for ceph_exporter, because it is easier for container deployment
* Make rados connection short lived - connection is established when sending a mon command, and closed after the command returns.
* Use mock library to implement noop connection for unit tests, which was super hard to maintain.
* Readme update

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module github.com/digitalocean/ceph_exporter
+
+go 1.15
+
+require (
+	github.com/beorn7/perks v0.0.0-20160804104726-4c0e84591b9a // indirect
+	github.com/ceph/go-ceph v0.0.0-20181217221554-e32f9f0f2e94
```
Comment (baergj) [inline go.mod:6]: Hmm, this seems like an older version of `go-ceph`, no? IIRC we had it at 0.5.0 on this branch previously.

File: collectors/conn.go [hunk 2]
```diff
@@ -155,180 +153,3 @@ func (c *RadosConn) GetPoolStats(pool string) (stat *rados.PoolStat, err error)
 
 	return
 }
-
-// NoopConn is the stub we use for mocking rados Conn. Unit testing
-// each individual collectors becomes a lot easier after that.
-// TODO: both output and cmdOut provide the command outputs for "go test", but
-// we can deprecate output, because cmdOut is able to hold the outputs we desire
-// for multiple commands for "go test".
-type NoopConn struct {
```
Comment (baergj) [inline collectors/conn.go:45]: Thank you thank you thank you for eliminating this monstrosity.

File: README.md [hunk 3]
```diff
@@ -1,40 +1,58 @@
...
+  the cluster (default `/etc/ceph/ceph.conf`)
+* `CEPH_USER`: a Ceph client user used to connect to the cluster (default
+  `admin`)
 
-We use Ceph's [official Golang client](https://github.com/ceph/go-ceph) to run commands on the cluster.
+We use Ceph's [official Golang client](https://github.com/ceph/go-ceph) to run
+commands on the cluster.
 
-Ceph exporter is tested only on Ceph's Hammer(v0.94) and Jewel(v10.2) releases. It might not work as expected with older or non-LTS versions of Ceph.
+The `cepp_exporter` is tested only on Ceph Nautilus releases. It might not
```
Comment (baergj) [inline README.md:33]: s/cepp/ceph/

Also, might want to specify `This ceph_exporter branch is tested ...`, just to make it clear that the user should look at other branches for other releases.

**Review by baergj — APPROVED** [review]

**Review by Matt1360 — APPROVED** [review]
Man, this update has it all:  Modules, updated prometheus client, updated copyright footer in the readme, improved logging.

:tada:

File: go.mod (hunk 1 above)
Comment (yuezhu) [reply go.mod:6]: Ah! Good catch! This's definitely off.

Updated to the `0.6.0` from `go get`

```
$ GO111MODULE=on go mod tidy                                   12:00:31.667
go: finding module for package github.com/ceph/go-ceph/rados
go: downloading github.com/ceph/go-ceph v0.6.0
go: found github.com/ceph/go-ceph/rados in github.com/ceph/go-ceph v0.6.0
```

**Comment by baergj** [comment]
Yeah, that copyright update will make a huge difference. ;)

File: README.md (hunk 3 above)
Comment (yuezhu) [reply README.md:33]: Sure, updated.

File: go.mod (hunk 1 above)
Comment (baergj) [reply go.mod]: :+1:

File: README.md (hunk 3 above)
Comment (baergj) [reply README.md]: :+1:

**Pull request merged**